### PR TITLE
Pip Libraries as Datasets

### PR DESCRIPTION
**Pull request by Isaac-Flath**
@jph00 

I added 4 functions to make uploading pip libraries to kaggle datasets easier.  Many competitions require no internet mode for inference, so having an easy way to create and update libraries as datasets is convenient.

Let me know if there’s any adjustments that should be made, or if this isn’t functionality that you would like to live in fastkaggle.

### CODE REVIEW

File: fastkaggle/core.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 # %% auto 0
-__all__ = ['iskaggle', 'import_kaggle', 'setup_comp', 'nb_meta', 'push_notebook']
+__all__ = ['iskaggle', 'import_kaggle', 'setup_comp', 'nb_meta', 'push_notebook', 'mk_dataset', 'get_dataset', 'get_pip_library',
+           'push_dataset']
 
 # %% ../00_core.ipynb 3
-import os,json
+import os,json,subprocess
 from fastcore.utils import *
+from kaggle import api
```
Comment (jph00) [inline fastkaggle/core.py]: Use `import_kaggle()` to import the api, and only do it when needed, since otherwise it breaks when run on Kaggle.

**Review by jph00 — CHANGES_REQUESTED** [review]
Great stuff! Just one request for changes...

File: fastkaggle/core.py (hunk 1 above)
Comment (Isaac-Flath) [reply fastkaggle/core.py]: I have adjusted this in the notebook and also made that change in a couple already existing functions.

**Comment by jph00** [comment]
Thank you @Isaac-Flath !

**Comment by jph00** [comment]
BTW I just noticed you didn't add any docs - maybe you could do another PR sometime with some explanations of what this is, and how to use it in practice, and what problem it solves?

**Comment by Isaac-Flath** [comment]
Ok I will add documentation. 

i'd like the end state to have a function that takes a `requirements.txt` file and uploads each library to kaggle datasets in the `requirements.txt` file automatically.  I think it'd work in roughly this way:
1. Loop through each library in `requirements.txt`
2. Check for existing kaggle dataset, create if doesn't exist or download if it does exist
3. Compare version in kaggle dataset to latest version in pip

I think if a version is specified it would create a dataset with that particular version in the name.  If a version is not specified it would just create a dataset with no version in the name that could be continuously updated to the latest version.

Thoughts?

**Comment by Isaac-Flath** [comment]
Unfortunately I have lost the ability to update datasets via either the CLI or python API.  There's an open issue where someone was experiencing the issue that I commented on but it's been there a while with no movement and the kaggle api github repo seems pretty stagnant, so I am not optimistic about getting resolution.  I get success messages and even hyperlinks to the alleged updated dataset from the kaggle api, and can even see a processing error bar in the kaggle website UI, but nothing actually gets created or updated anymore.

Sadly since I can't actually test anything anymore - I'm going to have to stop work on adding this capability to fastkaggle.

**Pull request merged**